pkg/cli/admin/release/extract_tools.go

```diff
@@ -1286,3 +1287,32 @@ func newIncluder(config manifestInclusionConfiguration) includer {
 		return m.Include(config.ExcludeIdentifier, config.RequiredFeatureSet, config.Profile, config.Capabilities, config.Overrides)
 	}
 }
+
+// ManifestReceiver has a TarEntryCallback function which can be used to as a callback to ExtractOptions.TarEntryCallback.
+// It feeds the downstream manifestsCallback
+// * with the manifests from every file whose name is not in skipNames, OR
+// * with the reader that contains the content of each file whose name is in skipNames.
+// All the errors encountered when parsing the manifests are collected in ManifestErrs.
+type ManifestReceiver struct {
+	manifestsCallback func(filename string, manifests []manifest.Manifest, reader io.Reader) (cont bool, err error)
+	skipNames         sets.Set[string]
+
+	ManifestErrs []error
+}
+
+func (mr *ManifestReceiver) TarEntryCallback(h *tar.Header, _ extract.LayerInfo, r io.Reader) (cont bool, err error) {
+	if mr.skipNames.Has(h.Name) {
+		return mr.manifestsCallback(h.Name, nil, r)
+	}
+
+	if ext := path.Ext(h.Name); len(ext) == 0 || !(ext == ".yaml" || ext == ".yml" || ext == ".json") {
+		return true, nil
+	}
+	klog.V(4).Infof("Found manifest %s", h.Name)
+	ms, err := manifest.ParseManifests(r)
+	if err != nil {
+		mr.ManifestErrs = append(mr.ManifestErrs, errors.Wrapf(err, "error parsing %s", h.Name))
```

Comment:
This makes the callback concurrent-unsafe; that's fine (IIRC you already checked once that it is never called in parallel - but maybe it should?) but the GoDoc should mention it.